TestingApp/DeeplinkService/DeeplinkService.swift

```diff
@@ -0,0 +1,59 @@
+import UIKit
+
+protocol DeeplinkHandler {
+    init(window: UIWindow?)
+    func handleDeeplink(
+        with path: String?,
+        queryItems: [URLQueryItem]?
+    ) -> Bool
+}
+
+final class DeeplinksService {
+    enum Host: String {
+        case configure, widgets
+    }
+
+    private let window: UIWindow?
+    private let handlers: [Host: DeeplinkHandler.Type]
+
+    init(
+        window: UIWindow?,
+        handlers: [Host: DeeplinkHandler.Type]
+    ) {
+        self.window = window
+        self.handlers = handlers
+    }
+}
+
+extension DeeplinksService {
+    @discardableResult
+    func openUrl(
+        _ url: URL,
+        withOptions options: [UIApplication.OpenURLOptionsKey: Any]
+    ) -> Bool {
+        return handleDeepLink(url)
+    }
+}
+
+// MARK: - Private methods
+
+private extension DeeplinksService {
+    @discardableResult
+    func handleDeepLink(_ link: URL) -> Bool {
+        guard let components = URLComponents(url: link, resolvingAgainstBaseURL: false),
```

Comment:
This bit of logic could be covered with a test as well, since a lot of conditions have to be met to return `true`.

Reply:
Added tests bedcfffdb2dc2d9b2c8f19c96d83ce597a5ecc97